src/app/server/Server.h

```diff
@@ -192,6 +192,8 @@ struct ServerInitParams
     // If the ICD Check-In protocol use-case is supported and no strategy is provided, server will use the default strategy.
     app::ICDCheckInBackOffStrategy * icdCheckInBackOffStrategy = nullptr;
 #endif // CHIP_CONFIG_ENABLE_ICD_CIP
+
+    chip::app::DataModel::Provider * dataModelProvider = nullptr;
```

Comment:
This needs to be documented as "must be non-null for a valid ServerInitParams" or something like that.